core/src/main/java/net/mabako/steamgifts/adapters/viewholder/GiveawayListItemViewHolder.java

```diff
@@ -136,6 +140,35 @@ public void onError() {
         indicatorLevelNegative.setVisibility(giveaway.isLevelNegative() ? View.VISIBLE : View.GONE);
         indicatorPrivate.setVisibility(giveaway.isPrivate() ? View.VISIBLE : View.GONE);
         indicatorRegionRestricted.setVisibility(giveaway.isRegionRestricted() ? View.VISIBLE : View.GONE);
+
+        // Initialize the enter button
+        // Check if logged or the quick enter button setting is enabled
+        boolean loggedIn = SteamGiftsUserData.getCurrent(null).isLoggedIn();
+        if (!loggedIn || !PreferenceManager.getDefaultSharedPreferences(fragment.getContext()).getBoolean("preference_giveaway_show_quick_enter", false)){
+            giveawayEnterButton.setVisibility(View.GONE);
+        } else {
+            // Set the correct text based on the giveaway is already entered or not
+            if (giveaway.isEntered()) {
+                giveawayEnterButton.setText(activity.getString(R.string.leave_giveaway));
```

Comment:
I would prefer if this was an icon from font-awesome. The list of hidden games is an example where a "x" is used for the quick action, I think, which I feel is more appropriate. 

This issue likely is worse on low resolution devices.